artifactory_cleanup/rules/delete.py

```diff
@@ -92,37 +93,85 @@ class delete_empty_folder(Rule):
     """
 
     def _aql_add_filter(self, aql_query_list):
-        update_dict = {
-            "repo": {
-                "$match": "deleteEmptyFolder",
-            }
-        }
-        aql_query_list.append(update_dict)
+        # Get list of all files and folders
+        all_files_dict = {"path": {"$match": "**"}, "type": {"$eq": "any"}}
+        aql_query_list.append(all_files_dict)
         return aql_query_list
 
     def _filter_result(self, result_artifact):
-        r = self.artifactory_session.get(
-            "{}/api/repositories?type=local".format(self.artifactory_server)
-        )
-        r.raise_for_status()
-        repositories = r.json()
-
-        for count, repository in enumerate(repositories, start=1):
-            if repository["packageType"] == "GitLfs":
-                # GitLfs should be handled by the jfrog cli: https://jfrog.com/blog/clean-up-your-git-lfs-repositories-with-jfrog-cli/
-                print(
-                    f"Skipping '{repository['key']}' because it is a Git LFS repository"
-                )
-                continue
-
-            url = "{}/api/plugins/execute/deleteEmptyDirsPlugin?params=paths={}".format(
-                self.artifactory_server, repository["key"]
-            )
-
-            print(
-                f"Deleting empty folders for '{repository['key']}' - {count} of {len(repositories)}"
-            )
-            r = self.artifactory_session.post(url)
-            r.raise_for_status()
-
-        return []
+
+        # convert list of files to dict
+        # Source: https://stackoverflow.com/a/58917078
+
+        def nested_dict():
+            """
+            Creates a default dictionary where each value is an other default dictionary.
+            """
+            return defaultdict(nested_dict)
+
+        def default_to_regular(d):
+            """
+            Converts defaultdicts of defaultdicts to dict of dicts.
+            """
+            if isinstance(d, defaultdict):
+                d = {k: default_to_regular(v) for k, v in d.items()}
+            return d
+
+        def get_path_dict(artifacts):
```

Comment:
I think the function should have the link to the SO question and have a short description what it does

Reply:
This is now fixed in the refactoring